include/graaflib/graph.h

```diff
@@ -179,6 +179,16 @@ class graph {
    */
   [[nodiscard]] vertex_id_t add_vertex(auto&& vertex);
 
+  /**
+   * Add a vertex to the graph with a specific ID
+   *
+   * @param  vertex The vertex to be added
+   * @param  id The requested ID for the new vertex
+   * @return vertices_id_t - The ID of the new vertex
+   * @throws id_taken exception - If the relevant ID is already in use
+   */
+  [[nodiscard]] vertex_id_t add_vertex(auto&& vertex, vertex_id_t id);
```

Comment:
[very minor]: Since the user already knows the ID of the vertex, it is not necessary to mark the return value as `nodiscard` since the user can safely ignore it.
```suggestion
   * @throws id_taken exception - If the relevant ID is already in use
   */
  vertex_id_t add_vertex(auto&& vertex, vertex_id_t id);
```